src/states/adult/mod.rs

```diff
@@ -411,9 +410,9 @@ impl Adult {
         if self.in_location(msg.message_dst()) {
             let signed_msg = msg.take_or_deserialize_signed_routing_message()?;
             match &signed_msg.routing_message().content {
-                RoutingVariant::GenesisUpdate(info) => {
+                Variant::GenesisUpdate(info) => {
                     self.verify_signed_message(&signed_msg)?;
-                    return self.handle_genesis_update(info.clone());
+                    return self.handle_genesis_update((**info).clone());
                 }
                 _ => {
```

Comment:
hence the reason to enumerate all, it is strange that for the direct message we choose to do something different (i.e not add to backlog).